drivers/wifi/infineon/cyw43xxx_drv.c

```diff
@@ -0,0 +1,713 @@
+
+	/* Call WHD API to send out the Packet */
+	ret = whd_network_send_ethernet_data(cyw43xxx_if, (void *)buf);
+	if (ret != CY_RSLT_SUCCESS) {
+		LOG_ERR("whd_network_send_ethernet_data failed");
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+		data->stats.errors.tx++;
+#endif
+		return -EIO;
+	}
+
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+	data->stats.bytes.sent += pkt_len;
+	data->stats.pkts.tx++;
+#endif
+
+	return 0;
+}
+
+void cy_network_process_ethernet_data(whd_interface_t interface, whd_buffer_t buffer)
+{
+	struct net_pkt *pkt;
+	uint8_t *data = whd_buffer_get_current_piece_data_pointer(interface->whd_driver, buffer);
+	uint32_t len = whd_buffer_get_current_piece_size(interface->whd_driver, buffer);
+	bool net_pkt_unref_flag = false;
+
+	if ((interface->role == WHD_STA_ROLE) &&
+	    (net_if_flag_is_set(cyw43xxx_wifi_iface, NET_IF_UP))) {
+		if (cyw43xxx_wifi_iface == NULL) {
+			LOG_ERR("network interface unavailable");
+			cy_buffer_release(buffer, WHD_NETWORK_RX);
+			return;
+		}
+
+		pkt = net_pkt_rx_alloc_with_buffer(cyw43xxx_wifi_iface, len, AF_UNSPEC, 0,
+						   K_NO_WAIT);
+
+		if (pkt != NULL) {
+			if (net_pkt_write(pkt, data, len) < 0) {
+				LOG_ERR("Failed to write pkt");
+				net_pkt_unref_flag = true;
+			}
+
+			if ((net_pkt_unref_flag) || (net_recv_data(cyw43xxx_wifi_iface, pkt) < 0)) {
+				LOG_ERR("Failed to push received data");
+				net_pkt_unref_flag = true;
+			}
+		} else {
+			LOG_ERR("Failed to get net buffer");
+		}
+	}
+	/* Release a packet buffer */
+	cy_buffer_release(buffer, WHD_NETWORK_RX);
+
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+	cyw43xxx_wifi_data.stats.bytes.received += len;
+	cyw43xxx_wifi_data.stats.pkts.rx++;
+#endif
+
+	if (net_pkt_unref_flag) {
+		net_pkt_unref(pkt);
+#if defined(CONFIG_NET_STATISTICS_WIFI)
+		cyw43xxx_wifi_data.stats.errors.rx++;
+#endif
+	}
+}
+
+static void *link_events_handler(whd_interface_t ifp, const whd_event_header_t *event_header,
+				 const uint8_t *event_data, void *handler_user_data)
+{
+	ARG_UNUSED(ifp);
+	ARG_UNUSED(event_data);
+	ARG_UNUSED(handler_user_data);
+
+	(void)k_msgq_put(&cyw43xxx_wifi_msgq, event_header, K_FOREVER);
+	return NULL;
+}
+
+static void cyw43xxx_event_task(void)
+{
+	whd_event_header_t event_header;
+
+	while (1) {
+		(void)k_msgq_get(&cyw43xxx_wifi_msgq, &event_header, K_FOREVER);
+
+		switch ((whd_event_num_t)event_header.event_type) {
+		case WLC_E_LINK:
+			break;
+
+		case WLC_E_DEAUTH_IND:
+		case WLC_E_DISASSOC_IND:
+			break;
+
+		default:
+			break;
+		}
+	}
+}
+
+static void cyw43xxx_mgmt_init(struct net_if *iface)
+{
+	const struct device *dev = net_if_get_device(iface);
+	struct cyw43xxx_wifi_runtime *data = dev->data;
+	struct ethernet_context *eth_ctx = net_if_l2_data(iface);
+
+	eth_ctx->eth_if_type = L2_ETH_IF_TYPE_WIFI;
+	data->iface = iface;
+	cyw43xxx_wifi_iface = iface;
+
+	/* Start interface when we are actually connected with WiFi network */
+	(void)net_if_flag_set(iface, NET_IF_NO_AUTO_START);
```

Comment:
This is the reason you need to run `net iface up 1` on the shell sample, you are telling the interface to not start automatically. If you remove this it will come up automatically.

Reply:
Thanks! removed this line, now it auto up.